### DISCUSSION TITLE
pluralize function in laravel

### DISCUSSION THREAD

**Discussion by lorvent** [laravel/framework#36080] [Q&A]
Hello,
I tried searching for pluralize function and even though i can find function.... i am not able to the place where team has added exceptions like `child`

are we using any third party dependency for that?

---

Also, i want to use same plural method in javascript, when user inputs singular word, i want to show plural word.

i have found some packages like https://github.com/plurals/pluralize but some words are not matching exactly like laravel and i want to maintain same for both languages.

so do you suggest using an ajax request to find laravel one and return it OR use a js based package?

thanks

**Comment by Davidnadejdin** [comment]
Laravel uses https://github.com/doctrine/inflector, or rather complements it(for example to throw exceptions). More details `Illuminate\Support\Pluralizer`

**Reply by lorvent** [reply]
thanks for the response. i have searched `Illuminate\Support\Pluralizer` and can't find rules like child to children.

i can't find such rule in `inflector` too but probably they are using regex somewhere.

overall..do you suggest using a js equivalent or just making ajax request so it always matches with laravel?

thanks
